sdk/identity/identity/src/credentials/managedIdentityCredential.ts

```diff
@@ -120,12 +120,12 @@ export class ManagedIdentityCredential implements TokenCredential {
     }
 
     return {
-      url: process.env.MSI_ENDPOINT,
+      url: process.env.IDENTITY_ENDPOINT || process.env.MSI_ENDPOINT,
```

Comment:
We can't really treat these variables identically. The presents of the IDENTITY_SECRET indicates we have to use the later API version of the Managed Identity protocol, which has an impact on the request we send and the response we can expect to get back.

Reply:
@schaabs thank you for this great feedback!